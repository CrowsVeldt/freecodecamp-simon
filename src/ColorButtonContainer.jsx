import ColorButton from './ColorButton'
import React, { Component } from 'react'

class ColorButtonContainer extends Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  activateButton = (num) => {
    switch (num) {
      case 1:
        this.audioElement1.play()
        break;
      case 2:
        this.audioElement2.play()
        break;
      case 3:
        this.audioElement3.play()
        break;
      default:
        this.audioElement4.play()
        break;
    }    
  }

  render () {

    return (

      <div className='colorButtonContainer'>
        <ColorButton
          className='greenButton'
          green
          onClick={() => { this.activateButton(1) }}
        />
        <audio ref={(audio1) => { this.audioElement1 = audio1 }}>
          <source src={'https://s3.amazonaws.com/freecodecamp/simonSound1.mp3'} />
        </audio>

        <ColorButton
          className='redButton'
          red
          onClick={() => { this.activateButton(2) }}
        />
        <audio ref={(audio2) => { this.audioElement2 = audio2 }}>
          <source src={'https://s3.amazonaws.com/freecodecamp/simonSound2.mp3'} />
        </audio>

        <ColorButton
          className='blueButton'
          blue
          onClick={() => { this.activateButton(3) }}
        />
        <audio ref={(audio3) => { this.audioElement3 = audio3 }}>
          <source src={'https://s3.amazonaws.com/freecodecamp/simonSound3.mp3'} />
        </audio>

        <ColorButton
          className='yellowButton'
          yellow
          onClick={() => { this.activateButton(4) }}
        />
        <audio ref={(audio4) => { this.audioElement4 = audio4 }}>
          <source src={'https://s3.amazonaws.com/freecodecamp/simonSound4.mp3'} />
        </audio>
      </div>
    
    )
  
  }

}

export default ColorButtonContainer
