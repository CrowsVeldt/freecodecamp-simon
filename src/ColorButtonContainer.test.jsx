import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ColorButtonContainer from './ColorButtonContainer'

describe('the ColorButtonContainer component', () => {
  
  test('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ColorButtonContainer />, div)
  })

  test('renders correctly', () => {
    const tree = renderer.create(<ColorButtonContainer />)
    expect(tree).toMatchSnapshot()
  })

})
