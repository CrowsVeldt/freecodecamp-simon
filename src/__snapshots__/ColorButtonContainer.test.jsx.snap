// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ColorButtonContainer component renders correctly 1`] = `
<div
  className="colorButtonContainer"
>
  <button
    className="sc-bdVaJa ivxUti"
    onClick={[Function]}
  >
    <audio>
      <source
        src="https://s3.amazonaws.com/freecodecamp/simonSound1.mp3"
      />
    </audio>
  </button>
  <button
    className="sc-bdVaJa hBZMzp"
    onClick={[Function]}
  >
    <audio>
      <source
        src="https://s3.amazonaws.com/freecodecamp/simonSound2.mp3"
      />
    </audio>
  </button>
  <button
    className="sc-bdVaJa ihYLjM"
    onClick={[Function]}
  >
    <audio>
      <source
        src="https://s3.amazonaws.com/freecodecamp/simonSound3.mp3"
      />
    </audio>
  </button>
  <button
    className="sc-bdVaJa lfqBkn"
    onClick={[Function]}
  >
    <audio>
      <source
        src="https://s3.amazonaws.com/freecodecamp/simonSound4.mp3"
      />
    </audio>
  </button>
</div>
`;
